schedule_lessons/dashboard/static/dashboard/js/maps.js

```diff
@@ -0,0 +1,74 @@
+// This example adds a search box to a map, using the Google Place Autocomplete
+// feature. People can enter geographical searches. The search box will return a
+// pick list containing a mix of places and predicted search terms.
+
+// This example requires the Places library. Include the libraries=places
+// parameter when you first load the API. For example:
+// <script src="https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY&libraries=places">
+
+function initAutocomplete() {
+  var map = new google.maps.Map(document.getElementById('map'), {
```

Comment:
no vars please :) just use `const` or `let` to stick with better standards and ES6

Reply:
What's wrong with vars?

Reply:
many things. firstly they are out of date, we now have `const` and `let` that do the job better.
also, one huge problem with vars is hoisting and how un-debuggable it makes the code.

here's an example of how hoisting works in js when using vars:

example 1 (lets say in a language like java that does have typed variables)

```
public void printDavid() {
    String x = "hey david";
    System.out.println(x);
}
```
This code above outputs "hey david"

```
public void printDavid() {
    System.out.println(x);
    String x = "hey david";
}
```
The code above produces an exception because the String variable x has not been defined yet.

In Javascript (when using vars), let's see what happens:

Example:

```
function printDavid() {
    var x = "hey dave";
    console.log(x);
}
```
The code above prints "hey dave" to the console.

```
function printDave() {
    console.log(x);
    var x = "hey dave";
}
```
The code above SHOULD produce an error because you are trying to print a value of a variable within a function where it hasnt been defined yet - BUT because using vars js automatically "hoists" the var declaration to the top of the function and then it works! and prints "hey dave"

In this example, it may seem like a beneficial feature; however, in more complex code it can become a nightmare to debug and figure out why a value has a particular value and cause many headaches.

@AhmedNSidd @DSchriemer @Nppatel97

Reply:
Hope that makes things a bit more clear team 😸

Reply:
as to why I always am pushing for ES6 standards to be used in js files hah